### PR TITLE
DIRECTOR: Lingo fixes, add support for displaying 2-bit and 4-bit bitmaps

### PR DESCRIPTION
**Pull request by moralrecordings**

### CODE REVIEW

File: engines/director/castmember.cpp [hunk 1]
```diff
@@ -153,9 +155,6 @@ BitmapCastMember::BitmapCastMember(Cast *cast, uint16 castId, Common::SeekableRe
 		if (_bitsPerPixel == 0)
 			_bitsPerPixel = 1;
 
-		if (_bitsPerPixel == 1)
```
Comment (sev-) [inline engines/director/castmember.cpp:156]: This needs to be tested with The Apartment

File: engines/director/castmember.cpp [hunk 2]
```diff
@@ -255,6 +258,62 @@ Graphics::MacWidget *BitmapCastMember::createWidget(Common::Rect &bbox, Channel
...
+			// Convert indexed image to indexed palette
+			Movie *movie = g_director->getCurrentMovie();
+			Cast *cast = movie->getCast();
+			Score *score = movie->getScore();
+			// Get the current score palette
+			int currentPaletteId = score->getCurrentPalette() ? score->getCurrentPalette() : cast->_defaultPalette;
+			PaletteV4 *currentPalette = g_director->getPalette(currentPaletteId);
+			// First, check if the palettes are different
+			switch (_bitsPerPixel) {
+				// 1bpp - this is preconverted to 0x00 and 0xff, change nothing.
```
Comment (sev-) [inline engines/director/castmember.cpp:271]: This not indented correctly: switch() does not add an indent level

**Review by sev- — CHANGES_REQUESTED** [review]
Good stuff, but we have to make sure that D2-D3 is working. The release cycle is close and we should not break the supported games

File: engines/director/graphics.cpp [hunk 3]
```diff
@@ -352,14 +352,18 @@ void inkDrawPixel(int x, int y, int src, void *data) {
...
+			// Find the inverse of the colour and match it back to the palette if required
+			byte rSrc, gSrc, bSrc;
+			wm->decomposeColor<T>(src, rSrc, gSrc, bSrc);
+
+			*dst = wm->findBestColor(~rSrc, ~gSrc, ~bSrc);
 		}
 		break;
 	case kInkTypeTransparent:
-		*dst = p->applyColor ? (src & p->foreColor) | (*dst & ~src) : (*dst & ~src);
+		*dst = p->applyColor ? (~src | p->backColor) & (*dst | src) : *dst | src;
```
Comment (sev-) [inline engines/director/graphics.cpp:363]: We have bunch of tests here: https://github.com/scummvm/director-tests/tree/master/D3-mac

You may consider checking them (and extending the test suite :) )

**Comment by moralrecordings** [comment]
Oh my god I fell for the "pending" label again, thanks Github designers

**Comment by sev-** [comment]
I'd love to merge it before the release, but it still breaks several movies on our buildbot. Do you plan to look into that?

**Comment by sev-** [comment]
Thank you! Merging

**Pull request merged**